spec/controllers/registrations_controller_spec.rb

```diff
@@ -5,43 +5,90 @@
 describe Api::V1::RegistrationsController do
   render_views
 
+  let!(:user) { create(:user) }
+
   before(:each) do
     @request.env['devise.mapping'] = Devise.mappings[:user]
-    @user = FactoryGirl.create(:user)
   end
 
   describe "POST 'users/'" do
-    it 'allows to create an user' do
-      @attr = { username: 'test', email: 'test@test.com', password: '12345678', password_confirmation: '12345678' }
-      post :create, user: @attr, format: 'json'
-      parsed_response = JSON.parse(response.body)
-      expect(response.status).to eq 200
+    let(:username) { 'test' }
+    let(:email) { 'test@test.com' }
+    let(:password) { '12345678' }
+    let(:password_confirmation) { '12345678' }
+
+    let(:attrs) {
+      {
+        username: username,
+        email: email,
+        password: password,
+        password_confirmation: password_confirmation,
+      }
+    }
+
+    it 'returns a successful response' do
+      post :create, user: attrs, format: 'json'
+
+      expect(response.status).to eq(200)
+    end
+
+    it 'creates the user' do
+      post :create, user: attrs, format: 'json'
+
       new_user = User.find_by_email('test@test.com')
       expect(new_user).to_not be_nil
     end
 
-    it 'should not create an user with invalid email' do
-      @attr = { username: 'test', email: 'notanemail', password: '12345678', password_confirmation: '12345678' }
-      post :create, user: @attr, format: 'json'
-      new_user = User.find_by_email('test@test.com')
-      expect(new_user).to be_nil
-      expect(response.response_code).to eq 400
+    context 'when the email is not correct' do
+      let(:email) { 'invalid_email' }
+
+      it 'does not create a user' do
+        expect{ post :create, user: attrs, format: 'json' }.not_to change{ User.count }
+      end
+
+      it 'does not return a successful response' do
+        post :create, user: attrs, format: 'json'
+
+        expect(response.response_code).to eq(400)
```

Comment:
again it will be better to check the message as well, I will commit this kind of comments from now on. But keep in mind that for a successful response asking for status 200 will do, but since you can have different errors it is a good practice to check message on 40n codes.